src/init.c

```diff
@@ -3432,7 +3432,7 @@ enum xnn_status xnn_initialize(const struct xnn_allocator* allocator) {
     allocator = &xnn_default_allocator;
   }
   #ifdef _MSC_VER
-    _InterlockedCompareExchangePointer(&init_allocator, allocator, NULL);
+    _InterlockedCompareExchangePointer((void* volatile)(&init_allocator), allocator, NULL);
```

Comment:
Just `(void* volatile) &init_allocator`, no need for extra brackets.

Reply:
Also, [the argument in the function signature](https://docs.microsoft.com/en-us/cpp/intrinsics/interlockedcompareexchangepointer-intrinsic-functions?view=msvc-160) has `void* volatile*` type

Reply:
Should be fixed now